### PR TITLE
examples/console: add persistence

### LINKED ISSUE #227: sdk/agent, sdk/state: define a story for persistence
We need to decide how the SDK will support persistence.

My preference is we find an elegant way to let the application decide how to persist the state.

I think this would look like:
- Adding an event handler to the agent that returns a copy of the state whenever is has changed, or for the sake of safety, whenever the agent calls a function on the channel.
- Moving the changing state of a channel into a type that has all its fields exported so the state of the channel can be easily copied out and is accessible to the caller.

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add persistence to a simple JSON file to the `examples/console` example.

### Why
To exercise the functionality adding snapshotting (persistence) that was added in #273.

Close #227

### CODE REVIEW

File: examples/console/main.go [hunk 1]
```diff
@@ -72,88 +95,131 @@ func run() error {
...
+			SequenceNumberCollector:    horizon,
+			BalanceCollector:           horizon,
+			Submitter:                  submitter,
+			Streamer:                   horizon,
+			EscrowAccountKey:           escrowAccountKey,
+			EscrowAccountSigner:        signerKey,
+			LogWriter:                  os.Stderr,
+			Events:                     events,
+		}
+		if filename != "" {
```
Comment (acharb) [inline examples/console/main.go:213]: 💭  feels weird to me that there could be a case where file exists but filename is empty. Which would leave the config with an empty Snapshotter field.

Thoughts on erroring if filename isn't given on the cli? Alternatively adding a default filename name at the top eg. "demo-snapshot.json"

**Review by acharb — COMMENTED** [review]
💯  just had 1 💭

File: examples/console/main.go (hunk 1 above)
Comment (leighmcculloch) [reply examples/console/main.go]: Thanks for catching this, the `filename != ""` condition is unnecessary. If there's a `file` there is always a `filename`.

**Review by leighmcculloch — COMMENTED** [review]
Addressed feedback in 5d72ec4.

**Review by acharb — APPROVED** [review]
lgtm!

**Pull request merged**